engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -206,6 +206,34 @@ public JobRecord updateJobRetries(final long jobKey, final int retries) {
     return job;
   }
 
+  @Override
+  public void cleanupTimeoutsWithoutJob() {
+    deadlinesColumnFamily.whileTrue(
+        (key, value) -> {
+          final var jobKey = key.second().inner();
+          final var deadline = key.first().getValue();
+          final var job = jobsColumnFamily.get(jobKey);
+          if (job == null || job.getRecord().getDeadline() != deadline) {
```

Comment:
:question: I'm not entirely sure if this is safe. Could there be a case where the deadline of this job is not in the deadline column family and we would end up deleting all deadlines for this job?

Reply:
we need this case as there could be multiple entries from the previous logic and the new logic only deletes one exact entry, right?

Reply:
Yes, exactly.

Reply:
The safest thing to do would be to completely clear the deadline column family and recreate it based on deadlines found for each job.

Reply:
AFAIK there isn't a case where an activated job wouldn't have an entry in the deadline column family. Every time a job is activated, its deadline and key are added to the deadline column family.

> The safest thing to do would be to completely clear the deadline column family and recreate it based on deadlines found for each job.

It's safest, but it's probably more expensive since we would be going through all the available jobs instread of the activated jobs subset.